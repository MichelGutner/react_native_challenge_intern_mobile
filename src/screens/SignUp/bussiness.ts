import { useNavigation } from '@react-navigation/native';
import { doCreateUser } from './../../services/loginAPI';



type Props = {
  username: string;
  email: string;
  password: string;
  onSuccess: () => void;
  onError: () => void;
};

export const doCreateRequest = async ({
  username,
  email,
  password,
  onSuccess,
  onError,
}: Props) => {
  const result = await doCreateUser({
    username: username,
    email: email,
    password: password,
  });
  if (result === 'OK') {
    onSuccess();
  } else {
    onError();
  }
};

