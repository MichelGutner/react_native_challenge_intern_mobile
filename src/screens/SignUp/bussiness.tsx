import { useNavigation } from '@react-navigation/native';
const navigation = useNavigation();

export function handleSignIn() {
  navigation.reset({
    routes: [{ name: 'SignIn' }],
  });
}
