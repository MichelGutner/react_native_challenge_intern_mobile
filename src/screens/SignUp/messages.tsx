import React from 'react';
import styled from 'styled-components/native';
import { Themes } from '../../../themes/themes';

type Props = {
  initialText: string;
  registerText: string;
  finishText: string;
};

export const finishRegister = 'Finalizar Cadastro';

export const Messages = ({ initialText, registerText, finishText }: Props) => {
  return (
    <>
      <MessageText>{initialText}</MessageText>
      <RegisterText>{registerText}</RegisterText>
      <MessageText>{finishText}</MessageText>
    </>
  );
};

const MessageText = styled.Text`
  font-size: 13px;
  font-weight: bold;
  color: ${Themes.colors.whiteOpacity};
`;
const RegisterText = styled.Text`
  font-weight: bold;
  color: ${Themes.colors.white};
`;
