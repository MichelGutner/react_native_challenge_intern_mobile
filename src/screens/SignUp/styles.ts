import { Themes } from './../../../themes/themes';
import styled from 'styled-components/native';

export const Container = styled.SafeAreaView`
  flex: 1;
  background-color: ${Themes.colors.backgroundSanar};
`;

export const ImageLogoSignUp = styled.Image`
  position: absolute;
  width: 152px;
  height: 45px;
  left: 104px;
  top: 96px;
`;

export const InputAreaView = styled.View`
  position: absolute;
  left: 24px;
  top: 213px;
`;

export const ButtonRegister = styled.TouchableOpacity`
  position: absolute;
  width: 312px;
  height: 48px;
  left: 24px;
  top: 421px;
  border-radius: 4px;
  background-color: ${Themes.colors.white};
  align-items: center;
  justify-content: center;
`;

export const TitleButtonRegister = styled.Text`
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;

  display: flex;
  align-items: center;
  text-align: center;

  color: ${Themes.colors.backgroundSanar};
`;

export const RegisterTextAlert = styled.Text`
  position: absolute;
  width: 296px;
  height: 45px;
  left: 32px;
  top: 485px;

  font-family: Red Hat Display;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;

  text-align: center;

  color: ${Themes.colors.whiteOpacity};
`;
