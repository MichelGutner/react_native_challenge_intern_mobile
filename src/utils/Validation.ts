const regx = /[^@ \t\r\n]+@[^@ \t\r\n]+\.[^@ \t\r\n]+/;

export const validationEmail = (email: string) => {
  if (regx.test(email)) {
    return true;
  } else {
    return false;
  }
};

const validationPasswordLenght = (password: string) => {
  if (password.length < 5) {
    return false;
  } else {
    return true;
  }
};

const validationPasswordHasOneUpperCaseLetter = (password: string) => {
  return true;
};

export const validationPassword = (password: string) => {
  const isPasswordLenghtValid = validationPasswordLenght(password);
  const hasPasswordUpperCaseLetter =
    validationPasswordHasOneUpperCaseLetter(password);
  return isPasswordLenghtValid && hasPasswordUpperCaseLetter;
};

// bloco password
//   if (password === null || password === '') {
//     let letterUpper = false;
//     let letterLower = false;
//     let haveNumber = false;
//     let haveSpecialCharacter = false;

//     // Mínimo de 8 caracteres.
//     if (password.length < 8) {
//       listErrors.push('A senha precisa de pelo menos 8 caracteres');
//     }

//     // Pelo menos 1 caractere maiúsculo.
//     // Pelo menos 1 caractere minúsculo.
//     // Pelo menos 1 dígito.
//     for (let i = 0; i < password.length; i++) {
//       // UpperCase
//       const character = password.charAt(i);
//       if (
//         character === character.toUpperCase() &&
//         character !== character.toLowerCase()
//       )
//         letterUpper = true;

//       // LowerCase
//       if (
//         character === character.toLowerCase() &&
//         character !== character.toUpperCase()
//       )
//         letterLower = true;

//       // IsNumeric
//       if (!isNaN(parseFloat(character)) && isFinite(parseInt(character))) {
//         haveNumber = true;
//       }

//       if (!/^[a-zA-Z0-9]+$/.test(character)) {
//         haveSpecialCharacter = true;
//       }
//     }

//     if (!letterUpper)
//       listErrors.push('A senha precisa de pelo menos 1 caracter maiusculo');

//     if (!letterLower)
//       listErrors.push('A senha precisa de pelo menos 1 caracter minusculo');

//     if (!haveNumber) {
//       listErrors.push('A senha precisa de pelo menos 1 digito');
//     }

//     if (haveSpecialCharacter) {
//       listErrors.push('A senha precisa de pelo menos 1 caracter especial');
//     }
//   }
//   return listErrors;
// };
