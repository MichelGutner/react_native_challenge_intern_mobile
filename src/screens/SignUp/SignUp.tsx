import { useNavigation } from '@react-navigation/native';
import React, { useState } from 'react';
import { Alert } from 'react-native';
import { Themes } from '../../../themes/themes';
import { InputValidation } from '../../components/InputValidation/InputValidation';
import Loading from '../../components/loading/Loading';
import { doCreateRequest, onPressRegisterButtonGoBack } from './bussiness';
import { finishRegister, Messages } from './messages';
import {
  ButtonRegister,
  Container,
  ImageLogoSignUp,
  InputAreaView,
  RegisterTextAlert,
  TitleButtonRegister,
} from './styles';
import { validationEmail, validationPassword } from '../../utils/Validation';

const SignUp = () => {
  const navigation = useNavigation();
  const [username, setUserName] = useState('');
  const [password, setPassword] = useState('');
  const [email, setEmail] = useState('');
  const [loading, setLoading] = useState(false);

  const onPressRegisterButtonGoBack = () => {
    navigation.navigate('SignIn');
  };

  const onSuccess = () => {
    setLoading(false);
    Alert.alert(
      'Sucesso',
      'Clique em OK para ser redirecionado para tela de login',
      [{ text: 'OK', onPress: onPressRegisterButtonGoBack }]
    );
  };
  const onError = () => {
    setLoading(false);
  };

  const onSignUpPressButton = () => {
    setLoading(true);
    {
      if (!validationEmail(email)) {
        Alert.alert('Insira um email válido');
        setLoading(false);
        return;
      }
      if (!validationPassword(password)) {
        Alert.alert('Insira uma senha válida');
        setLoading(false);
        return;
      }
      doCreateRequest({
        username,
        email,
        password,
        onSuccess,
        onError,
      });
    }
  };

  return (
    <Container>
      <ImageLogoSignUp
        source={require('../../components/SanarLogo/Logo-negativo.png')}
      />
      <InputAreaView>
        <InputValidation
          value={username}
          onChangeText={(text: string) => {
            setUserName(text);
          }}
          secureText={false}
          isInputValid={false}
          placeholder={'Digite seu nome'}
          placeholderTextColor={Themes.colors.white}
        />
        <InputValidation
          value={email}
          onChangeText={(text: string) => {
            setEmail(text);
          }}
          secureText={false}
          isInputValid={validationEmail(email)}
          placeholder={'Digite seu e-mail'}
          placeholderTextColor={Themes.colors.white}
        />
        <InputValidation
          value={password}
          onChangeText={(text: string) => {
            setPassword(text);
          }}
          secureText={true}
          isInputValid={validationPassword(password)}
          placeholder={'Digite sua senha'}
          placeholderTextColor={Themes.colors.white}
        />
      </InputAreaView>
      <ButtonRegister onPress={onSignUpPressButton}>
        <TitleButtonRegister>{finishRegister}</TitleButtonRegister>
      </ButtonRegister>
      <RegisterTextAlert>
        <Messages
          initialText={'Ao clicar em'}
          registerText={' "Finilizar Cadastro" '}
          finishText={'você estará aceitando também nossos termos e condições.'}
        />
      </RegisterTextAlert>
      <Loading visible={loading} />
    </Container>
  );
};

export default SignUp;
