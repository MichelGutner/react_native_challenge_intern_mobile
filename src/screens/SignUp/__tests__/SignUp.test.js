import { NavigationContainer } from '@react-navigation/native';
import React from 'react';
import renderer from 'react-test-renderer';
import { validationEmail, validationPassword } from '../../../utils/Validation';
import SignUp from '../SignUp';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <NavigationContainer>
        <SignUp />
      </NavigationContainer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('should returns false when email is invalid', () => {
  const result = validationEmail('michel-gutner');
  expect(result).toBe(false);
});

test('should returns true when email is valid', () => {
  const result = validationEmail('michel-gutner@gmail.com');
  expect(result).toBe(true);
});
test('should returns false when password is invalid', () => {
  const result = validationPassword('123456789');
  expect(result).toBe(false);
});
test('should returns true false email is valid', () => {
  const result = validationPassword('Gutner@654');
  expect(result).toBe(true);
});
