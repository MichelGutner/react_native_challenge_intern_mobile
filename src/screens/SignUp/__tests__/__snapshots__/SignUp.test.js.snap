// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#0050F0",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/components/SanarLogo/Logo-negativo.png",
      }
    }
    style={
      Array [
        Object {
          "height": 45,
          "left": 104,
          "top": 96,
          "width": 152,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "left": 24,
          "top": 172,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Digite seu nome"
        placeholderTextColor="#FFF"
        secureTextEntry={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#0050f0",
              "borderColor": "#ffffff",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "white",
              "display": "flex",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "height": 48,
              "marginTop": 16,
              "textAlign": "center",
              "width": 312,
            },
          ]
        }
        value=""
      />
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "red",
              "fontSize": 20,
            },
            Object {
              "marginTop": 20,
            },
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Digite seu e-mail"
        placeholderTextColor="#FFF"
        secureTextEntry={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#0050f0",
              "borderColor": "#ffffff",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "white",
              "display": "flex",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "height": 48,
              "marginTop": 16,
              "textAlign": "center",
              "width": 312,
            },
          ]
        }
        value=""
      />
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "red",
              "fontSize": 20,
            },
            Object {
              "marginTop": 20,
            },
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Digite sua senha"
        placeholderTextColor="#FFF"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#0050f0",
              "borderColor": "#ffffff",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "white",
              "display": "flex",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "height": 48,
              "marginTop": 16,
              "textAlign": "center",
              "width": 312,
            },
          ]
        }
        value=""
      />
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "red",
              "fontSize": 20,
            },
            Object {
              "marginTop": 20,
            },
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 4,
        "height": 48,
        "justifyContent": "center",
        "left": 24,
        "opacity": 1,
        "top": 213,
        "width": 312,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignItems": "center",
            "color": "#0050F0",
            "display": "flex",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "500",
            "textAlign": "center",
          },
        ]
      }
    >
      Finalizar Cadastro
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "height": 60,
          "left": 32,
          "textAlign": "center",
          "top": 224,
          "width": 296,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(255,255,255,0.5)",
            "fontSize": 13,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Ao clicar em
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontWeight": "bold",
          },
        ]
      }
    >
       "Finilizar Cadastro" 
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(255,255,255,0.5)",
            "fontSize": 13,
            "fontWeight": "bold",
          },
        ]
      }
    >
      você estará aceitando também nossos termos e condições.
    </Text>
  </Text>
  <Modal
    hardwareAccelerated={false}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
    visible={false}
  />
</RCTSafeAreaView>
`;
